### PR TITLE
CSM/SSM: Add on_mods_loaded callback

### PR DESCRIPTION
**Pull request by nerzhul**
Added a new callback called just after mods finished loading.

Implementation is both client side and server side. Added a builtin callback showing mods are loaded in the logs too (we don't have it it can be nice too :) ), you can test it looking for your logs :)

### CODE REVIEW

**Comment by paramat** [comment]
For #7390

File: doc/lua_api.txt [hunk 1]
```diff
@@ -2793,6 +2793,8 @@ Call these functions only at load time!
 
 * `minetest.register_globalstep(func(dtime))`
     * Called every server step, usually interval of 0.1s
+* `minetest.register_on_mods_loaded(func())`
+    * Called just after mods have finished loading.
```
Comment (SmallJoker) [inline doc/lua_api.txt]: Could do well with a more detailed information when exactly it is called:
* Called after mods have finished loading and before the media is cached or the aliases handled.

File: builtin/game/init.lua [hunk 2]
```diff
@@ -34,3 +34,7 @@ dofile(gamepath.."forceloading.lua")
 dofile(gamepath.."statbars.lua")
 
 profiler = nil
+
+core.register_on_mods_loaded(function()
+	core.log("Server mods loaded.")
```
Comment (SmallJoker) [inline builtin/game/init.lua]: Seems superfluous. The mod loading procedure is already logged as verbose (or info?).

Comment (nerzhul) [reply builtin/game/init.lua]: i can remove it if wanted, at least it permits to show it works as intended :)
Do i need to keep it for CSM mods ?

File: builtin/client/init.lua [hunk 3]
```diff
@@ -21,3 +21,7 @@ core.register_on_formspec_input(function(formname, fields)
 		core.send_respawn()
 	end
 end)
+
+core.register_on_mods_loaded(function()
+	core.log("Client mods loaded.")
+end)
```
Comment (SmallJoker) [inline builtin/client/init.lua]: Same here as with the SSM builtin part. It however might fit better into the preview CSM mod.

**Pull request merged**